nan_persistent_12_inl.h

```diff
@@ -93,28 +101,57 @@ class Global : public v8::Global<T> {
 #else
 template<typename T>
 class Global : public v8::UniquePersistent<T> {
+  struct RValue {
+    inline explicit RValue(v8::UniquePersistent<T> *obj) : object_(obj) {}
+    v8::UniquePersistent<T> *object_;
+  };
+
  public:
   inline Global() : v8::UniquePersistent<T>() {}
 
-  template<typename S> inline Global(v8::Local<S> that) :
+  template<typename S> inline explicit Global(v8::Local<S> that) :
     v8::UniquePersistent<T>(v8::Isolate::GetCurrent(), that) {}
 
   template<typename S>
-  inline Global(const v8::PersistentBase<S> &that) :
+  inline explicit Global(const v8::PersistentBase<S> &that) :
       v8::UniquePersistent<S>(v8::Isolate::GetCurrent(), that) {}
 
   inline void Reset() { v8::PersistentBase<T>::Reset(); }
 
-  template <typename S>
+  template<typename S>
   inline void Reset(const v8::Local<S> &other) {
     v8::PersistentBase<T>::Reset(v8::Isolate::GetCurrent(), other);
   }
 
-  template <typename S>
+  template<typename S>
   inline void Reset(const v8::PersistentBase<S> &other) {
     v8::PersistentBase<T>::Reset(v8::Isolate::GetCurrent(), other);
   }
 
+#if NODE_MODULE_VERSION == NODE_0_12_MODULE_VERSION
+  inline void Empty() {
+    reinterpret_cast<v8::Persistent<T>*>(this)->ClearAndLeak();
```

Comment:
I can also add that `v8::Persistent::ClearAndLeak` was moved to `v8::PersistentBase::Empty` after V8 @ Node 0.12, but already in V8 @ io.js 1.0. It was changed to no longer return the encapsulated `T*`. However, since `v8::Persistent` is a `v8::PersistentBase` and `v8::UniquePersistent` is a `v8::PersistentBase` and the sole `val_` member resides in `v8::PersistentBase`, this is correct (in the loosest sense of correct). Nothing is virtual.